0000-api-versions.md

```diff
@@ -0,0 +1,50 @@
+# Meta
+[meta]: #meta
+- Name: API Versions
+- Start Date: 2019-08-05
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Begin versioning the various APIs that the spec describes. Then require these versions be specified in related TOML files, under an `api` field.
+
+# Motivation
+[motivation]: #motivation
+
+To enable logic and/or validation based on `api`. For instance, we could keep users from attempting to use a combination of incompatible buildpacks and lifecycle.
+
+# What it is
+[what-it-is]: #what-it-is
+
+There would be two pieces to this:
+
+1. A reference to the current versions of all APIs placed in the spec. Currently, we would version the Buildpack API (buildpack-to-lifecycle contract) and the Platform API (lifecycle-to-platform contract).
```

Comment:
Does it make sense that there's a distribution version? If we're going to break it down does a version per markdown file make sense?

Reply:
I could see that making sense eventually. Do we want to wait until/if we have a breaking change in the distribution spec?

Reply:
I think it should be mentioned as we expect to do this in the future.  However, with the no value implies 1.0, no one would be required to do anything about it today.